synedrion/src/cggmp21/sigma/mod_.rs

```diff
@@ -135,34 +136,37 @@ impl<P: SchemeParams> ModProof<P> {
         }
     }
 
-    pub fn verify(
-        &self,
-        rng: &mut impl CryptoRngCore,
-        pk: &PublicKeyPaillier<P::Paillier>,
-        aux: &impl Hashable,
-    ) -> bool {
+    pub fn verify(&self, pk: &PublicKeyPaillier<P::Paillier>, aux: &impl Hashable) -> bool {
         let challenge = ModChallenge::new(pk, &self.commitment, aux);
         if challenge != self.challenge {
             return false;
         }
 
+        let mut reader = XofHasher::new_with_dst(b"P_mod RNG").chain(aux).finalize_to_reader();
+        let mut seed = <ChaCha8Rng as SeedableRng>::Seed::default();
+        reader.read(&mut seed);
+        let mut rng = ChaCha8Rng::from_seed(seed);
+
```

Comment:
Is the idea here is to make the verification repeatable, is that why we want a seeded RNG? Given how few inputs we use to feed the `XofHasher`, figuring out the rng state is perhaps fairly easy, is that a concern?

Reply:
Yes, `verify()` cannot be RNG-dependent in order to create verifiable evidences. I suppose we can feed `commitment` and `pk` to the hash as well.

I don't think it's a concern. Also, if you recall the current primality check is MR base 2 - Lucas - MR random base, where only the last step uses RNG, and that last step is never actually hit in tests. It's only there because GMP does that.